src/core/Makefile.am

```diff
@@ -183,6 +184,7 @@ noinst_HEADERS                      = \
     crypto/aes_ecb.hpp                \
     crypto/hmac_sha256.hpp            \
     crypto/mbedtls.hpp                \
+    crypto/pbkdf2_cmac.h              \
```

Comment:
You can add a new line [here](https://github.com/chshu/openthread/blob/115d7d38491dc245149a62dac66b2af713bd823e/third_party/mbedtls/mbedtls-config.h#L41) to disable that warning (4267).